hashes/src/internal_macros.rs

```diff
@@ -249,3 +249,14 @@ macro_rules! hash_type_no_default {
     };
 }
 pub(crate) use hash_type_no_default;
+
+macro_rules! incomplete_block_len_impl {
+    () => {
+        fn incomplete_block_len(&self) -> usize {
+            let block_size = BLOCK_SIZE as u64; // Cast usize to u64 is ok.
+            // After modulo operation we know cast u64 to usize as ok.
+            (<Self as crate::HashEngine>::n_bytes_hashed(&self) % block_size) as usize
```

Comment:
This has an undocumented precondition that `BLOCK_SIZE` is not defined as `u64` with value > `usize::MAX`. :) I think it's OK.

Reply:
Added #3315